### PR TITLE
FRS-908 - Add CI to PR's

### PR DESCRIPTION
**Pull request by footerz**
## Changes
- Adds package json
- Adds stylelint
- Adds stylesheet check for CI
- Adds temp base variables for UFC and Golf
- Adds new PR CI workflow

## Reference
Fixes [FRS-908]

[FRS-908]: https://imgarenahub.atlassian.net/browse/FRS-908

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,12 +8,18 @@ Stylesheets must be named as follows - operator name and theme name combined wit
 
 E.g. operator-theme.css
 
-This will result in a final external url of https://static.imgarena.com/event-centre-styles/operator-theme.css
+This will result in a final external url of https://static.imgarena.com/event-centre-styles/golf/operator-theme.css
+
+The parent directory of the stylesheet will change depending on what event centre directory you place your stylesheet.
```
Comment (MrTibbles) [inline README.md:13]: IMO this is not of interest to the intended audience of this README.

File: README.md [hunk 2]
```diff
@@ -8,12 +8,18 @@ Stylesheets must be named as follows - operator name and theme name combined wit
...
-Place stylesheets in the /stylesheets directory
+Place stylesheets in relevant event centre directory within the /stylesheets directory.
 
 ## How to upload
 
 Simply raise a PR for your new stylesheet. Once approved and merged this will automatically be uploaded to the correct endpoint.
+
+## Mandatory variables
+
+The /mandatory-variables directory houses the css variable names which MUST be present in each event centres stylesheet. This will be automatically checked for each PR raised.
```
Comment (MrTibbles) [inline README.md:25]: This contradicts the implementation of variables being optionally overridden. If a variable is not overridden then it's value will be taken from the default values defined in the base.css file which is within each event repo. 
The variables are not mandatory, they are **available to be overridden**.

File: scripts/stylesheet-check.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+const { mandatoryGolfVariables } = require('../mandatory-variables/golf');
+const { mandatoryUfcVariables } = require('../mandatory-variables/ufc');
+
+const fs = require('fs');
+const { readdirSync } = require('fs');
+const readline = require('readline');
+
+const variablesIncluded = (arr, target) => target.every(v => arr.includes(v));
```
Comment (MrTibbles) [inline scripts/stylesheet-check.js:8]: Prefer meaningful variable names, please amend the use of `v`

File: scripts/stylesheet-check.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+directories.forEach(directory => {
+    fs.readdir('stylesheets/' + directory, (err, files) => { 
+        if (err) 
+          console.log(err); 
+        else { 
+          files.forEach(file => {
+            const includedVariables = [];
+    
+            async function processLineByLine() {
+                const rl = readline.createInterface({
```
Comment (MrTibbles) [inline scripts/stylesheet-check.js:23]: Prefer meaningful variable names, please amend the use of `rl`

File: scripts/stylesheet-check.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+                    input: fs.createReadStream('stylesheets/' + directory + '/' + file),
+                    crlfDelay: Infinity
+                });
+    
+                for await (const line of rl) {
+                    if (line.includes('--') && !line.includes('/*') && !line.includes('*/')) {
```
Comment (MrTibbles) [inline scripts/stylesheet-check.js:29]: Consider using a regex pattern, would be neater IMO and faster

File: scripts/stylesheet-check.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+                        const variable = line.substring(line.lastIndexOf("-") + 1,line.lastIndexOf(":"));
+        
+                        includedVariables.push(variable);
+                    };
+                };
+                
+                if (!variablesIncluded(includedVariables, directory === 'golf' ? mandatoryGolfVariables : mandatoryUfcVariables)) {
+                    console.log('FAILURE: stylesheets/' + directory + '/' + file + ' is missing a mandatory ' + directory + ' css variable(s).');
```
Comment (MrTibbles) [inline scripts/stylesheet-check.js:37]: Again, they are not mandatory, they are available to be overridden. As discussed, this script is sense checking the variables being passed in to check that only available variables are being used and they are spelt correctly. NOT that they are all being overridden.

**Review by MrTibbles — CHANGES_REQUESTED** [review]

File: scripts/stylesheet-check.js (hunk 6 above)
Comment (footerz) [reply scripts/stylesheet-check.js:37]: Ok, so essentially we're just spell checking any supplied variables against the base ones. I was on a different page.

File: README.md (hunk 1 above)
Comment (footerz) [reply README.md]: Sure. f28ad95

File: README.md (hunk 2 above)
Comment (footerz) [reply README.md]: Sure! dc9c2d8

File: scripts/stylesheet-check.js (hunk 6 above)
Comment (footerz) [reply scripts/stylesheet-check.js]: Here we go dd13e55

File: scripts/stylesheet-check.js (hunk 3 above)
Comment (footerz) [reply scripts/stylesheet-check.js]: This no longer exists dd13e55

File: scripts/stylesheet-check.js (hunk 4 above)
Comment (footerz) [reply scripts/stylesheet-check.js]: Changed dd13e55

File: scripts/stylesheet-check.js (hunk 5 above)
Comment (footerz) [reply scripts/stylesheet-check.js:26]: Politely declining. 😂

File: event-centre-variables/golf.js [hunk 7]
```diff
@@ -0,0 +1,3 @@
+const mandatoryGolfVariables = ['white', 'grey'];
+
+module.exports = mandatoryGolfVariables;
```
Comment (MrTibbles) [inline event-centre-variables/golf.js:3]: again, they are not mandatory, they are "supportedVariables"

File: event-centre-variables/ufc.js [hunk 8]
```diff
@@ -0,0 +1,3 @@
+const mandatoryUfcVariables = ['white', 'black'];
```
Comment (MrTibbles) [inline event-centre-variables/ufc.js:1]: same as above

File: scripts/stylesheet-check.js (hunk 5 above)
Comment (MrTibbles) [reply scripts/stylesheet-check.js:26]: lols

**Review by MrTibbles — CHANGES_REQUESTED** [review]
For consistency and clarity please remove references to the variables being mandatory, as they are not.

File: event-centre-variables/ufc.js (hunk 8 above)
Comment (footerz) [reply event-centre-variables/ufc.js]: 430178a

File: event-centre-variables/golf.js (hunk 7 above)
Comment (footerz) [reply event-centre-variables/golf.js]: 430178a

**Review by MrTibbles — APPROVED** [review]

**Pull request merged**